File: 0011-container-with-most-water/0011-container-with-most-water.py
class Solution:
    def maxArea(self, height: List[int]) -> int:
        answer = 0
        left = 0
        right = len(height)-1
        
        while (left < right):
            current_height = 0
        
            if height[right] > height[left]:
                current_height = height[left]
            else:
                current_height = height[right]
            answer = max(answer, current_height * (right - left ))
            
            if height[right] > height[left]:
                left += 1
            else:
                right -= 1
        return answer 
            

File: LeetCode/0011-container-with-most-water/0011-container-with-most-water.java
class Solution {
    public int maxArea(int[] height) {
        int left = 0;
        int right = height.length - 1;
        int maxArea = 0;
        //현재  left와 right를 비교
        while(left < right) {
            maxArea = Math.max(maxArea, (right - left) * Math.min(height[left], height[right]));
            if(height[left] <= height[right]) {
                left++;
            } else{
                right--;
            }
        }

        return maxArea;
    }
}